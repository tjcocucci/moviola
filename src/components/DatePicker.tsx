import React, { useState } from 'react';
import ReactDatePicker from 'react-datepicker';
import 'react-datepicker/dist/react-datepicker.css';

export default function DatePicker({
  handleChange,
}: {
  handleChange: Function;
}) {
  const [date, setDate] = useState<Date | null>(null);

  const handleDateChange = (d: Date) => {
    setDate(d);
    handleChange(d);
  };

  return (
    <ReactDatePicker
      data-testid="date-picker"
      selected={date}
      onChange={handleDateChange}
      className="p-2 border border-gray-300 rounded w-2/3"
      dateFormat="dd/MM/yyyy"
    />
  );
}
