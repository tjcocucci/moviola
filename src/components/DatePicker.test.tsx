import { describe, it, expect, vi } from 'vitest';
import { fireEvent, getByTestId, render } from '@testing-library/react';
import DatePicker from './DatePicker';
import ReactDatePicker from 'react-datepicker';
import 'react-datepicker/dist/react-datepicker.css';

const defaultProps = {
  handleChange: () => {},
};

describe('[ components / DatePicker ]', () => {
  describe('when component is rendered', () => {
    it('should render a date picker', () => {
      const { asFragment } = render(<DatePicker {...defaultProps} />);
      expect(asFragment()).toMatchSnapshot();
    });
  });
  describe('when date is not changed', () => {
    it('should not call handleChange', () => {
      const props = {
        handleChange: vi.fn(),
      };
      expect(props.handleChange).not.toHaveBeenCalled();
    });
  });
  describe('when date is changed', () => {
    it('should call handleChange', () => {
      const props = {
        handleChange: vi.fn(),
      };
      const { container } = render(<DatePicker {...props} />);
      const inputElement = container.querySelector('input');

      if (inputElement !== null) {
        fireEvent.change(inputElement, {
          target: { value: '2020-01-01' },
        });
      } else {
        throw new Error('Input element not found');
      }

      expect(props.handleChange).toHaveBeenCalled();
    });
  });
});
