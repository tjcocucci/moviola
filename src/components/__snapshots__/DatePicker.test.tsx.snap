// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`[ components / DatePicker ] > when component is rendered > should render a date picker 1`] = `
<DocumentFragment>
  <div
    class="react-datepicker-wrapper"
  >
    <div
      class="react-datepicker__input-container "
    >
      <span
        aria-live="polite"
        class="react-datepicker__aria-live"
        role="alert"
      />
      <input
        class="p-2 border border-gray-300 rounded w-2/3"
        type="text"
        value=""
      />
    </div>
  </div>
</DocumentFragment>
`;
